stdlib/public/Concurrency/CMakeLists.txt

```diff
@@ -49,16 +49,21 @@ if(NOT swift_concurrency_async_fp_mode)
   set(swift_concurrency_async_fp_mode "always")
 endif()
 
-# Don't emit extended frame info on platforms other than darwin, system
-# backtracer and system debugger are unlikely to support it.
-if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
-  list(APPEND SWIFT_RUNTIME_CONCURRENCY_C_FLAGS
-    "-fswift-async-fp=${swift_concurrency_async_fp_mode}")
-  list(APPEND SWIFT_RUNTIME_CONCURRENCY_SWIFT_FLAGS
-    "-Xfrontend"
-    "-swift-async-frame-pointer=${swift_concurrency_async_fp_mode}")
+if(XCODE)
+  message(WARNING "The -fswift-async-fp flag is being disabled because of an "
+                  "issue with the Xcode build, see SR-15216 for info.")
 else()
-  list(APPEND SWIFT_RUNTIME_CONCURRENCY_C_FLAGS "-fswift-async-fp=never")
+  # Don't emit extended frame info on platforms other than darwin, system
+  # backtracer and system debugger are unlikely to support it.
+  if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
```

Comment:
Hey @buttaface 
Does this works locally for you? I thought about testing that but because of this comment https://github.com/apple/swift/pull/39368#issuecomment-924167811, trying to compile the stdlib without this flag doesn't work so I'm not sure this will solve the issue.

Reply:
As I mentioned a couple times on that earlier pull, I don't use macOS, so this pull has not been tested whatsoever.

My understanding of the problem from what you and others initially described is that the host clang at `/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/` is being wrongly used by the Xcode generator to build the Swift stdlib, despite the Swift CMake config specifying that it use the just-built Swift-forked clang instead. This pull works around that by not passing the flag when the Xcode generator is used, because the host clang doesn't support this flag.

In your linked comment, you then claim that the Xcode generator was using the Swift-forked, freshly-built clang to build this Concurrency library, in which case this pull won't work, as you then have much deeper problems if the Xcode generator is randomly choosing which clang compiler to use.

I suggest you do a normal build without this pull, and paste the full error you get here. Then, apply this pull and rebuild without cleaning and see what difference it makes. If both runs show different compilers being used, I don't know what to tell you, there's something deeply wrong with your Xcode setup.

Reply:
> I suggest you do a normal build without this pull, and paste the full error you get here. Then, apply this pull and rebuild without cleaning and see what difference it makes. If both runs show different compilers being used, I don't know what to tell you, there's something deeply wrong with your Xcode setup.

Thank you for the suggestion, I'll take a deeper look at it

Reply:
Hi @buttaface,
No, I tried this change and keep hitting that linker issue ... so I tried Doug's change locally from https://github.com/apple/swift/pull/39573 and it seems to be enough to unblock Xcode and build everything =]

Reply:
OK, can you provide the error messages I asked for before, ie the full error on a stock build and after applying this patch? You don't need to do a full rebuild because these patches only affect the stdlib, ie this should be pretty easy to do with an existing build just by changing the patch applied to this file.

Reply:
Does Xcode not show the failing command? I don't care about the error as much as what particular command caused it.

Reply:
It fails while linking `libswift_Concurrency.dylib` 
```
Ld ${SRC_DIR}/build/Xcode-RelWithDebInfoAssert+swift-DebugAssert/swift-macosx-arm64/Debug/lib/swift/macosx/arm64/libswift_Concurrency.dylib normal (in target 'swift_Concurrency-macosx-arm64' from project 'Swift')
    cd ${SRC_DIR}/swift
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -target arm64-apple-macos11.3 -dynamiclib -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk -L${SRC_DIR}/build/Xcode-RelWithDebInfoAssert+swift-DebugAssert/swift-macosx-arm64/Debug/lib/swift/macosx/arm64 -L${SRC_DIR}/build/Xcode-RelWithDebInfoAssert+swift-DebugAssert/llvm-macosx-arm64/RelWithDebInfo/lib/Debug -L${SRC_DIR}/build/Xcode-RelWithDebInfoAssert+swift-DebugAssert/llvm-macosx-arm64/RelWithDebInfo/lib -L${SRC_DIR}/build/Xcode-RelWithDebInfoAssert+swift-DebugAssert/swift-macosx-arm64/Debug/lib/swift/macosx/arm64/Debug -L${SRC_DIR}/build/Xcode-RelWithDebInfoAssert+swift-DebugAssert/swift-macosx-arm64/Debug/lib/swift/macosx/arm64 -L${SRC_DIR}/build/Xcode-RelWithDebInfoAssert+swift-DebugAssert/swift-macosx-arm64/Debug/bin/../lib/swift/macosx/arm64/Debug -L${SRC_DIR}/build/Xcode-RelWithDebInfoAssert+swift-DebugAssert/swift-macosx-arm64/Debug/bin/../lib/swift/macosx/arm64 -L${SRC_DIR}/build/Xcode-RelWithDebInfoAssert+swift-DebugAssert/swift-macosx-arm64/Debug/bin/../lib/swift/macosx/Debug -L${SRC_DIR}/build/Xcode-RelWithDebInfoAssert+swift-DebugAssert/swift-macosx-arm64/Debug/bin/../lib/swift/macosx -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/lib/swift/Debug -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/lib/swift -F${SRC_DIR}/build/Xcode-RelWithDebInfoAssert+swift-DebugAssert/swift-macosx-arm64/Debug/lib/swift/macosx/arm64 -filelist ${SRC_DIR}/build/Xcode-RelWithDebInfoAssert+swift-DebugAssert/swift-macosx-arm64/stdlib/public/Concurrency/Swift.build/Debug/swift_Concurrency-macosx-arm64.build/Objects-normal/arm64/swift_Concurrency.LinkFileList -install_name /usr/lib/swift/libswift_Concurrency.dylib -Xlinker -object_path_lto -Xlinker ${SRC_DIR}/build/Xcode-RelWithDebInfoAssert+swift-DebugAssert/swift-macosx-arm64/stdlib/public/Concurrency/Swift.build/Debug/swift_Concurrency-macosx-arm64.build/Objects-normal/arm64/swift_Concurrency_lto.o -stdlib\=libc++ -target arm64-apple-macosx10.9 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk -arch arm64 -F/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/../../../Developer/Library/Frameworks -Wl,-sectcreate,__TEXT,__info_plist,${SRC_DIR}/build/Xcode-RelWithDebInfoAssert+swift-DebugAssert/swift-macosx-arm64/stdlib/public/Concurrency/Info.plist -Wl,-application_extension -Xlinker -compatibility_version -Xlinker 1 -dynamiclib -Wl,-headerpad_max_install_names ${SRC_DIR}/build/Xcode-RelWithDebInfoAssert+swift-DebugAssert/swift-macosx-arm64/Debug/lib/swift/macosx/arm64/libswiftCore.dylib -Xlinker -no_adhoc_codesign -Xlinker -dependency_info -Xlinker ${SRC_DIR}/build/Xcode-RelWithDebInfoAssert+swift-DebugAssert/swift-macosx-arm64/stdlib/public/Concurrency/Swift.build/Debug/swift_Concurrency-macosx-arm64.build/Objects-normal/arm64/swift_Concurrency_dependency_info.dat -o ${SRC_DIR}/build/Xcode-RelWithDebInfoAssert+swift-DebugAssert/swift-macosx-arm64/Debug/lib/swift/macosx/arm64/libswift_Concurrency.dylib
```
As for the build-script command I'm using `SKIP_XCODE_VERSION_CHECK=1 ./swift/utils/build-script --xcode --release-debuginfo --debug-swift --skip-build-benchmarks  --swift-darwin-supported-archs "$(uname -m)" --sccache  --skip-ios --skip-tvos --skip-watchos`